spec/formatting.md

```diff
@@ -324,8 +324,6 @@ The result of resolving _option_ values is a mapping of string identifiers to va
 For each _option_:
 
 - Resolve the _identifier_ of the _option_.
-- If the _option_'s _identifier_ already exists in the resolved mapping of _options_,
-   emit a _Duplicate Option Name_ error.
```

Comment:
_Duplicate Option Name_ **_is_** a data model error?

Reply:
The current formatting spec language assumes that it's dealing with a _valid_ message, i.e. one with no data model errors. [Duplicate Option Name](https://github.com/unicode-org/message-format-wg/blob/main/spec/errors.md#duplicate-option-name) errors are primarily defined in `errors.md`, but for messages coming from syntax, they're also defined here: https://github.com/unicode-org/message-format-wg/blob/e76196481b23e6e9245923a1239282e19484efd0/spec/syntax.md?plain=1#L567-L568
and as with all other data model errors they're required to be detected and emitted before formatting: https://github.com/unicode-org/message-format-wg/blob/e76196481b23e6e9245923a1239282e19484efd0/spec/errors.md?plain=1#L11-L14

So in order to get to this point in formatting, the current spec text must presume that this error has already been checked, and that therefore the condition for triggering it can't happen. So this statement should be removed to remove the current uncertainty it presents.

We also explicitly only allow for Resolution and Formatting errors to be ignored in expressions that aren't being formatted, and _not_ data model errors: https://github.com/unicode-org/message-format-wg/blob/e76196481b23e6e9245923a1239282e19484efd0/spec/errors.md?plain=1#L19-L21

Reply:
Thanks @eemeli. You're right, we do have that text in `errors.md`, etc.

I guess what I'm grappling with is: did we make the right choice here. @catamorphism and @mihnita might weigh in about how they've implemented the checks required by data model (that is, when do they check for duplicate options or missing variant keys or duplicate declarations). I'm fine with blowing up the message for all of these with the possible exception of duplicate option names (and even there I still want to blow up)

Reply:
Like I mentioned in #703 (just repeating it here to make the threads easier to follow) my implementation does a "static checking" pass for data model errors that are not checked during parsing, which is consistent with what @eemeli is saying.